src/main/java/at/ac/tuwien/kr/alpha/api/externals/stdlib/AspStandardLibrary.java

```diff
@@ -0,0 +1,157 @@
+import at.ac.tuwien.kr.alpha.api.externals.Predicate;
+import at.ac.tuwien.kr.alpha.common.terms.ConstantTerm;
+import at.ac.tuwien.kr.alpha.common.terms.Terms;
+
+/**
+ * Collection of methods that can be used as external atoms from ASP programs.
+ * Provides commonly used functionality such as basic string operations,
+ * datetime handling etc.
+ * 
+ * All functions exposed by this class are guaranteed to be stateless and
+ * side-effect free.
+ * 
+ * Copyright (c) 2020, the Alpha Team.
+ */
+public final class AspStandardLibrary {
+
+	private AspStandardLibrary() {
+
+	}
+
+	/**
+	 * Parses a string representing a datetime without time-zone and returns the
+	 * year, month, day, hours, minutes and seconds as separate symbolic integer
+	 * terms.
+	 * Example:
+	 * 
+	 * <pre>
+	 * 		A valid ground instance of the atom &stdlib_datetime_parse[DTSTR, "dd.mm.yyyy hh:MM:ss"](YEAR, MONTH, DAY, HOUR, MIN, SEC)
+	 * 		would be: &stdlib_datetime_parse["20.05.2020 01:19:13", "dd.mm.yyyy hh:MM:ss"](2020, 5, 20, 1, 19, 13)
+	 * </pre>
+	 * 
+	 * Timezones are not supported by this function. Datetime values are parsed
+	 * using {@link LocalDateTime.parse}.
+	 * 
+	 * @param datetime a string representing a datetime without time zone
+	 *                 information
+	 * @param format   a format string that is accepted by {@link DateTimeFormatter}
+	 * @return a 6-value integer tuple of format (YEAR, MONTH, DAY, HOUR, MIN, SEC)
+	 */
+	@Predicate
+	public static Set<List<ConstantTerm<Integer>>> stdlib_datetime_parse(String dtstr, String format) {
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern(format);
+		LocalDateTime datetime = LocalDateTime.parse(dtstr, formatter);
+		List<ConstantTerm<Integer>> terms = Terms.asTermList(
+				datetime.getYear(), datetime.getMonth().getValue(), datetime.getDayOfMonth(),
+				datetime.getHour(), datetime.getMinute(), datetime.getSecond());
+		return ExternalUtils.wrapAsSet(terms);
+	}
+
+	/**
+	 * Compares two datetimes and returns true iff the first datetime (dt1) is
+	 * before the second datetime (dt2). Both datetimes are represented as six
+	 * integers each, referring to years, months, days, hours, minutes and seconds
+	 * respectively.
+	 * 
+	 * @param dt1Year   the year field for dt1
+	 * @param dt1Month  the month field for dt1
+	 * @param dt1Day    the day field for dt1
+	 * @param dt1Hour   the hour field for dt1
+	 * @param dt1Minute the minute field for dt1
+	 * @param dt1Second the second field for dt1
+	 * @param dt2Year   the year field for dt2
+	 * @param dt2Month  the month field for dt2
+	 * @param dt2Day    the day field for dt2
+	 * @param dt2Hour   the hour field for dt2
+	 * @param dt2Minute the minute field for dt2
+	 * @param dt2Second the second field for dt2
+	 * @return true if dt1 is before dt2 in time, false otherwise
+	 */
+	@Predicate
+	public static boolean stdlib_datetime_is_before(int dt1Year, int dt1Month, int dt1Day, int dt1Hour, int dt1Minute, int dt1Second,
```

Comment:
I always had the idea in mind that calling Java methods on objects as external atoms should be hassle free. In this case, my preference would be to obtain two `LocalDateTime` objects an then call `isBefore` in some generic way, therefore eliminating the need to for a special "stdlib_datetime_is_before" implementation.

Reply:
In principle, I agree. However, I don't see that working in Alpha in the immediate future...
I think in a perfect world, I'd be able to have `LocalDateTime` as a "type" in ASP on which the "<"-operator is defined and mapped to the java implementation of `LocalDateTime#isBefore`.

Without any kind of ASP type system, I'd rather not directly call methods on objects from ASP, it just seems to be inviting chaos (side-effects, symmetry of comparison operations, etc..). While all that can also happen with my version from above, writing a dedicated method at least forces people to think about what a method is doing before exposing it to the ASP solver.